cmd/systray/systray.go

```diff
@@ -55,6 +59,25 @@ type Menu struct {
 	eventCancel func() // cancel eventLoop
 }
 
+func init() {
+	desktop := strings.ToLower(os.Getenv("XDG_CURRENT_DESKTOP"))
+	switch desktop {
+	case "kde":
+		// KDE doesn't need a delay, and actually won't render submenus
+		// if we delay for more than about 400µs.
+		newMenuDelay = 0
+	default:
+		// Add a slight delay to ensure the menu is created before adding items.
+		//
+		// Systray implementations that use libdbusmenu sometimes process messages out of order,
+		// resulting in errors such as:
+		//    (waybar:153009): LIBDBUSMENU-GTK-WARNING **: 18:07:11.551: Children but no menu, someone's been naughty with their 'children-display' property: 'submenu'
+		//
+		// See also: https://github.com/fyne-io/systray/issues/12
+		newMenuDelay = 100 * time.Millisecond
```

Comment:
skip this if GOOS != "linux" ?

Reply:
ah, good call.  I'm not discounting that I may also need hacks for mac/win later :)